# I aquired data from the TCGA data freeze which has combined all the data together from the various plaforms

# Methylation Patient Centric Table
# To generate DNA methylation calls for each sample per gene per overlapping platforms (HM27, HM450), we began by first collapsing multiple CpGs to one representative gene. 
# Using the associated gene expression data (organized as one gene - one expression value per sample), we merged the samples and CpG probes with gene expression data for each 
# platform. We next calculated the spearman correlation (rho) across all samples for all CpG probes for each gene to one gene expression value. For multiple CpGs for each 
# annotated gene promoter, we selected one CpG probe with the lowest correlation rho value to the associated gene expression profile to capture the most biologically 
# representative event (epigenetic silencing). This effectively reduced the number of CpG probes from N:1 to 1:1. Our data set was then reduced down to 279 samples x 9,453 
# CpG:Gene (HM27) and 74 samples x 11,040 CpG:Gene (HM450) with an overlap of 9,222 genes. Next, we assigned discrete categories based on the spearman correlation rho value 
# according to the following criteria:
    
#     1. Strongly negatively correlated (SNC) when rho value is less than 0.5;
# 2. Weakly negatively correlated (WNC) when rho value is between 0.5 and 0.25;
# 3. No negative correlation (NNC) when rho value is greater than 0.25.
# Next, we assigned samples to either the 10th (T10 or N10) or 90th (T90 or N90) percentile based on the observed beta-value across
# tumor samples (T) and normal samples (N). For the normal samples, we used four nontumor brain samples previously profiled on the HM27 platform (Noushmehr et al., 2010) 
# and 24 randomly selected normal samples across three different normal tissues (72 samples) profiled on the HM450 platform. The three different normal tissues selected 
# for this analysis were generated by TCGA for breast (BRCA), kidney (KIRC) and lung (LUSC) tumor studies.
# We assigned labels for each gene per platform per tissue type (tumor and normal) according to the following rules:
#     1. If percentile 90 < 0.25, we assign it as CUN or CUT (constitutively unmethylated in normal or tumor);
# 2. If percentile 10 > 0.75, we assign it as CMN or CMT (constitutively methylated in normal or tumor);
# 3. If percentile 10 > 0.25 and percentile 90 < 0.75, we assign it as IMN or IMT (intermediate methylated in normal or tumor); 4. 
3 If it doesn’t fall in any of the above categories, it is assign VMN or VMT (variably methylated in normal or tumor).
# Next we assigned a ‘call’ and a confidence ‘score’ for each possible combinations (48) [3 (SNC, WNC, NNC) x 4 (CUN, CMN, VMN, IMN) x 4 (CUT, CMT, VMT, IMT)] per platform. 
# We created the following relationship for each call and score based on our interpre- tation of the most informative epigenetic event (e.g. promoter DNA hypermethylation 
# and low expression). Users should understand that the selection and criteria performed were done to the best of our knowledge at the time. 
# We felt most confident with calling epige- netically silenced events and this is reflected in the confidence score. 

# The methylation calls are as follows:
#     MG: Methylation gain compared to normal ML: Methylation loss compared to normal MT: Methylated in tumor
# UT: Unmethylated in tumor
# ES: Epigenetically silenced
# UC: Unable to make call
# Methylation class confidence scores vary from 0 (no call) to 4 (high confidence).



setwd("~/Documents/public-datasets/TCGA/methylation/")
list.files()

calls = read.delim("TCGA_GBM_dnameth_calls_20120112_ver3.txt")
scores = read.delim("TCGA_GBM_dnameth_scores_20120112_ver3.txt")
categories = read.delim("TCGA_GBM_dnameth_call_categories.txt")
data = read.delim("DNA.methylation.k6.txt")